Formula/pdmenu.rb

```diff
@@ -0,0 +1,31 @@
+class Pdmenu < Formula
+  desc "Full screen menuing system for Unix"
+  homepage "https://joeyh.name/code/pdmenu/"
+  url "http://sources.buildroot.net/pdmenu/pdmenu-1.3.6.tar.gz"
+  sha256 "dedd2a4a5719278b5e49041161990c2f20b5909818837542aaca01819f2c14eb"
+  license "GPL-2.0-only"
+
+  depends_on "coreutils"
+  depends_on "gettext"
+  depends_on "s-lang"
+
+  def install
+    # should be fixed in the upcoming 1.3.7
+    inreplace "autoconf/makeinfo.in",
+      "LIBS		= @LIBS@ $(EFENCE)",
+      "LIBS		= @LIBS@ @INTLLIBS@ $(EFENCE)"
+
+    # we cannot actually use "/usr/share/locale"
+    inreplace "Makefile",
+      "LOCALEDIR	= $(INSTALL_PREFIX)/usr/share/locale",
+      "LOCALEDIR	= $(INSTALL_PREFIX)#{prefix}/locale"
+
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/pdmenu -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hello,

Yes, I know. But it's very difficult, or quite impossible to do better. The program does three things: 

- prints its version
- shows the help 
- displays a screen menu.

It calls  [tcgetpgrp](https://linux.die.net/man/3/tcgetpgrp) :
```
src/keyboard.c:16            if (tcgetpgrp(0) == -1) return 033; /* Detect if the terminal went away. */                                      
```
that fails in standard input is not a terminal. I cannot pipe a fake response to obtain something (e.g `pdmenu < response > excepted` won't work). Or I didn't find the solution :(

Eventually
```bash
pdmenu < /dev/null
```
exits with 0 if it has successfully read the example menu file. And 1 it not.

If it is crippling, I cancel the PR.

Reply:
Well, I thought it was not possible to catch the output of the screen, but I found a convenient way.

`pdmenu` without any argument displays the example menu, then quit because of [tcgetpgrp](https://linux.die.net/man/3/tcgetpgrp) that fails if stdin is not a terminal.

So, I look for some strings sent to the terminal as a test.